lib/account_deleter.rb

```diff
@@ -69,10 +68,6 @@ def delete_standard_person_associations
     end
   end
 
-  def disconnect_contacts
```

Comment:
Actually `before_destroy` block in `AspectMembership` doesn't delete contacts if it's `sharing: true`. So it's not the same thing as `disconnect_contacts` in `AccountDeleter`. I have no idea why it worked for you, but it now fails with contacts not empty for me.

Here is my pry session output that shows what happens to a contact actually:
```
From: /home/senya/source/diaspora/app/models/aspect_membership.rb @ line 14 :

     9:   has_one :user, :through => :contact
    10:   has_one :person, :through => :contact
    11: 
    12:   before_destroy do
    13:     binding.pry
 => 14:     if self.contact && self.contact.aspects.size == 1
    15:       self.user.disconnect(self.contact)
    16:     end
    17:     true
    18:   end
    19: 

[2] pry(#<AspectMembership>)> self.contact.aspects.size
=> 1
[4] pry(#<AspectMembership>)> user.contacts
=> [#<Contact:0x0000000dfeff08
  id: 1105,
  user_id: 1139,
  person_id: 4295,
  created_at: Mon, 14 Aug 2017 01:56:40 UTC +00:00,
  updated_at: Mon, 14 Aug 2017 01:56:40 UTC +00:00,
  sharing: true,
  receiving: true>]
[5] pry(#<AspectMembership>)> next

From: /home/senya/source/diaspora/app/models/aspect_membership.rb @ line 15 :

    10:   has_one :person, :through => :contact
    11: 
    12:   before_destroy do
    13:     binding.pry
    14:     if self.contact && self.contact.aspects.size == 1
 => 15:       self.user.disconnect(self.contact)
    16:     end
    17:     true
    18:   end
    19: 
    20:   def as_json(opts={})

[5] pry(#<AspectMembership>)> next

From: /home/senya/source/diaspora/app/models/aspect_membership.rb @ line 17 :

    12:   before_destroy do
    13:     binding.pry
    14:     if self.contact && self.contact.aspects.size == 1
    15:       self.user.disconnect(self.contact)
    16:     end
 => 17:     true
    18:   end
    19: 
    20:   def as_json(opts={})
    21:     {
    22:       :id => self.id,

[6] pry(#<AspectMembership>)> user.contacts.reload
=> [#<Contact:0x0000000f50cfa8
  id: 1105,
  user_id: 1139,
  person_id: 4295,
  created_at: Mon, 14 Aug 2017 01:56:40 UTC +00:00,
  updated_at: Mon, 14 Aug 2017 01:57:25 UTC +00:00,
  sharing: true,
  receiving: false>]
```